eng/feeds/arm/tspconfig.yaml

```diff
@@ -6,6 +6,39 @@ options:
     emitter-output-dir: "{project-root}/.."
     azure-resource-provider-folder: "resource-manager"
     output-file: "{azure-resource-provider-folder}/{service-name}/{version-status}/{version}/openapi.json"
+  "@azure-tools/typespec-csharp":
+    package-dir: {package-name}
+    clear-output-folder: true
+    model-namespace: false
+    namespace: "{package-dir}"
+  "@azure-tools/typespec-python":
+    package-dir: "{{#normalizePackageName}}{{parameters.ServiceNamespace}}{{/normalizePackageName}}"
+    package-name: "{package-dir}"
+    generate-test: true
+    generate-sample: true
+  "@azure-tools/typespec-java":
+    package-dir: "{{#normalizePackageName}}{{parameters.ServiceNamespace}}{{/normalizePackageName}}"
+    namespace: com.{{#toLowerCase}}{{parameters.ServiceNamespace}}{{/toLowerCase}}
+    service-name: "{service-name}"
+  "@azure-tools/typespec-ts":
+    package-dir: "{{#normalizePackageName}}{{parameters.ServiceNamespace}}{{/normalizePackageName}}"
```

Comment:
I may lack of some contexts, will this tspconfig be handled by some tool to replace these place holder? Could you please describe it? I can't verify it since I don't know what will be generated from `{{#normalizePackageName}}`, `{{parameters.ServiceNamespace}}` and `{{/normalizePackageName}}`. 

But I can provide the following ts's emitter options rules based on [modular config sample](https://github.com/Azure/azure-rest-api-specs/blob/main/specification/contosowidgetmanager/Contoso.Management/tspconfig.yaml#L29) used by js doc.
e.g. add `name: contoso` in parameters

```
package-dir: "arm-{name}"
packageDetails:
      name: "@azure/{package-dir}"
```

Reply:
@raych1 do you know about these placeholders `{{#normalizePackageName}}`, `{{parameters.ServiceNamespace}}` and `{{/normalizePackageName}}`? When and how to handle the placeholders?

Reply:
@maririos I am also curious about what are the scenarios behind this PR?

Reply:
This is the template project which will scaffold a new project based on the selected template when run `tsp init` command. The purpose is to ensure the generated `tspconfig.yaml` by these templates have the recommended settings.

Reply:
I copied the format from the data plane template, but I can delete the values if they are confusing

Reply:
Updated to @wanlwanl recommendations